themes/default/static/programs/aws-ec2-vpc-resources-java/src/main/java/myproject/App.java

```diff
@@ -0,0 +1,122 @@
+
+            // Create a route table.
+            var routes = new RouteTable("routes", RouteTableArgs.builder()
+                .vpcId(vpc.id())
+                .routes(RouteTableRouteArgs.builder()
+                    .cidrBlock("0.0.0.0/0")
+                    .gatewayId(gateway.id())
+                    .build())
+                .build());
+
+            // Associate the route table with the public subnet.
+            var routeTableAssociation = new RouteTableAssociation("route-table-association", RouteTableAssociationArgs.builder()
+                .subnetId(subnet.id())
+                .routeTableId(routes.id())
+                .build());
+
+            // Create a security group allowing inbound access over port 80 and outbound access to anywhere.
+            var securityGroup = new SecurityGroup("security-group", SecurityGroupArgs.builder()        
+                .name("allow_public")
+                .vpcId(vpc.id())
+                .build());
+    
+            var allowPort80Ipv4 = new SecurityGroupIngressRule("allowPort80Ingress", SecurityGroupIngressRuleArgs.builder()        
+                .securityGroupId(securityGroup.id())
+                .cidrIpv4("0.0.0.0/0")
+                .fromPort(80)
+                .ipProtocol("tcp")
+                .toPort(80)
+                .build());
+
+            var allowAllTrafficIpv4 = new SecurityGroupEgressRule("allowAllTrafficEgress", SecurityGroupEgressRuleArgs.builder()        
+                .securityGroupId(securityGroup.id())
+                .cidrIpv4("0.0.0.0/0")
+                .ipProtocol("-1")
+                .build());
+
+            // Find the latest Amazon Linux 2 AMI.
+            var ami = Ec2Functions.getAmi(GetAmiArgs.builder()
+                .owners("amazon")
+                .mostRecent(true)
+                .filters(GetAmiFilterArgs.builder()
+                    .name("description")
+                    .values("Amazon Linux 2 *")
+                    .build())
+                .build()).applyValue(getAmiResult -> getAmiResult.id());
+
+            // Create and launch an Amazon Linux EC2 instance into the public subnet.
+            var instance = new Instance("instance", InstanceArgs.builder()
+                .ami(ami)
+                .instanceType("t3.nano")
+                .subnetId(subnet.id())
+                .vpcSecurityGroupIds(Output.all(securityGroup.id()))
+                .userData(
+                        "#!/bin/bash\n" +
+                        "sudo yum update -y\n" +
+                        "sudo yum upgrade -y\n" +
+                        "sudo amazon-linux-extras install nginx1 -y\n" +
+                        "sudo systemctl enable nginx\n" +
+                        "sudo systemctl start nginx;\n"
+                )
+                .build());
+
+            // Export the instance's publicly accessible URL.
+            ctx.export("instanceURL", instance.publicIp().applyValue(ip -> "http://" + ip));
```

Comment:
We can use a helper here to be consistent with the others:

```suggestion
            ctx.export("instanceURL", Output.format("http://%s", instance.publicIp()));
```